airbyte-integrations/connectors/source-jira/source_jira/client.py

```diff
@@ -92,3 +90,170 @@ def health_check(self) -> Tuple[bool, str]:
+            query_value = project_key.get(query_param)
+            for item in self.lists(name=f"project_{name}", url=f"/project/{query_value}/{url_name}", **ENTITIES_MAP.get(f"project_{name}")):
+                yield item
+
+    def stream__avatars(self, fields):
+        avatar_types = ("issuetype", "project", "user")
+        avatar_configs = ENTITIES_MAP.get("avatars")
+        for avatar_type in avatar_types:
+            for data in self.lists(name="avatars", url=f"/avatar/{avatar_type}/system", **avatar_configs):
+                yield data
+
+    def stream__filter_sharing(self, fields):
+        filter_sharing_configs = ENTITIES_MAP.get("filter_sharing")
+        for filter_item in self.lists(name="filters", **ENTITIES_MAP.get("filters")):
+            filter_item_id = filter_item.get("id")
+            for permission in self.lists(name="filter_sharing", url=f"/filter/{filter_item_id}/permission", **filter_sharing_configs):
+                yield permission
+
+    def stream__issue_comments(self, fields):
+        issue_keys = self._get_issue_keys()
+        for item in issue_keys:
+            issue_key = item.get("key")
+            for comment in self.lists(name="issue_comments", url=f"/issue/{issue_key}/comment", **ENTITIES_MAP.get("issue_comments")):
+                yield comment
+
+    def stream__issue_custom_field_contexts(self, fields):
+        for field in self._get_custom_fields():
+            for context in self.lists(
+                name="issue_custom_field_contexts",
+                url=f"/field/{field.get('id')}/context",
+                **ENTITIES_MAP.get("issue_custom_field_contexts"),
+            ):
+                yield context
+
+    def stream__issue_properties(self, fields):
+        issue_keys = self._get_issue_keys()
+        for item in issue_keys:
+            issue_key = item.get("key")
+            for issue_property_key_item in self.lists(
+                name="issue_property_keys", url=f"/issue/{issue_key}/properties", func=lambda v: v["keys"], params={}
+            ):
+                issue_property_key = issue_property_key_item.get("key")
+                for issue_property in self.lists(
+                    name="issue_properties",
+                    url=f"/issue/{issue_key}/properties/{issue_property_key}",
+                    **ENTITIES_MAP.get("issue_properties"),
+                ):
+                    yield issue_property
+
+    def stream__issue_remote_links(self, fields):
+        issue_keys = self._get_issue_keys()
+        for item in issue_keys:
+            issue_key = item.get("key")
+            for comment in self.lists(
+                name="issue_remote_links", url=f"/issue/{issue_key}/remotelink", **ENTITIES_MAP.get("issue_remote_links")
+            ):
+                yield comment
+
+    def stream__issue_votes(self, fields):
+        issue_keys = self._get_issue_keys()
+        for item in issue_keys:
+            issue_key = item.get("key")
```

Comment:
it seems we are always calling `_get_issue_keys` followed by `item.get("key")`. Why not DRY this inside `_get_issue_keys` thereby also saving space?